File: cpp/src/Drive/Tank.cpp
//#include <ctre/phoenix/MotorControl/DeviceCatalog.h>
//#include "ctre/phoenix/Drive/Tank.h"
//#include "HAL/DriverStation.h"
//
//namespace ctre { namespace drive {
//
//Tank::tank(ctre::Mechanical::Gearbox *left, ctre::Mechanical::Gearbox *right,
//		bool leftInvert, bool rightInvert)
//{
//    _left = left;
//    _right = right;
//
//    _left->SetInverted(leftInvert);
//    _right->SetInverted(rightInvert);
//}
//Tank::tank(ctre::motorcontrol::IMotorController *left, ctre::motorcontrol::IMotorController *right,
//		bool leftInvert, bool rightInvert)
//{
//    /* Create 2 single motor gearboxes */
//    ctre::Mechanical::Gearbox *temp1 = new ctre::Mechanical::Gearbox(left);
//    ctre::Mechanical::Gearbox *temp2 = new ctre::Mechanical::Gearbox(right);
//
//    _left = temp1;
//    _right = temp2;
//
//    _left->SetInverted(leftInvert);
//    _right->SetInverted(rightInvert);
//}
//void Tank::Set(Styles::Basic mode, float forward, float turn){
//    float l, r;
//	l = forward + turn;
//    r = forward - turn;
//
//    Drive(mode, l, r);
//}
//void Tank::SetVoltageRampRate(float rampRate){
//    _left->SetVoltageRampRate(rampRate);
//    _right->SetVoltageRampRate(rampRate);
//}
//void Tank::SetVoltageCompensationRampRate(float rampRate){
//    _left->SetVoltageCompensationRate(rampRate);
//    _right->SetVoltageCompensationRate(rampRate);
//}
//void Tank::ConfigPeakPercentOutputVoltage(float forwardVoltage, float reverseVoltage){
//    _left->ConfigPeakOutputVoltage(forwardVoltage, reverseVoltage);
//    _right->ConfigPeakOutputVoltage(forwardVoltage, reverseVoltage);
//}
//void Tank::ConfigNominalPercentOutputVoltage(float forwardVoltage, float reverseVoltage){
//    _left->ConfigNominalOutputVoltage(forwardVoltage, reverseVoltage);
//    _right->ConfigNominalOutputVoltage(forwardVoltage, reverseVoltage);
//}
//void Tank::Drive(Styles::Basic mode, float left, float right){
//    if (mode == Styles::Basic::VoltageBasic)
//    {
//        _left->SetControlMode(ctre::motorcontrol::ControlMode::BasicControlMode::kVoltageBasic);
//        _right->SetControlMode(ctre::motorcontrol::ControlMode::BasicControlMode::kVoltageBasic);
//    }
//    else if (mode == Styles::Basic::PercentOutputBasic)
//    {
//        _left->SetControlMode(ctre::motorcontrol::ControlMode::BasicControlMode::kPercentVbusBasic);
//        _right->SetControlMode(ctre::motorcontrol::ControlMode::BasicControlMode::kPercentVbusBasic);
//    }
//
//    _left->Set(left);
//    _right->Set(right);
//}
//
//}}

File: java/src/com/ctre/phoenix/drive/Tank.java
package com.ctre.phoenix.drive;
//package com.ctre.phoenix.Drive;
//
//import com.ctre.phoenix.Mechanical.*;
//import com.ctre.phoenix.motorcontrol.*;
//import com.ctre.phoenix.Drive.Styles;
//
//public class Tank implements IDrivetrain
//{
//	Gearbox _left;
//	Gearbox _right;
//	
//	public Tank(Gearbox left, Gearbox right, boolean leftInvert, boolean rightInvert)
//	{
//		GroupMotorControllers.Register(left.getMaster());
//		_left = left;
//		_right = right;
//		_left.setInverted(leftInvert);
//		_right.setInverted(rightInvert);
//	}
//	
//	public Tank(IMotorController m1, IMotorController m2, boolean leftInvert, boolean rightInvert)
//    {
//        GroupMotorControllers.Register(m1);
//        /* Create 2 single motor gearboxes */
//        Gearbox temp1 = new Gearbox(m1);
//        Gearbox temp2 = new Gearbox(m2);
//
//        _left = temp1;
//        _right = temp2;
//
//        _left.setInverted(leftInvert);
//        _right.setInverted(rightInvert);
//    } 
//	
//	/** Inherited from IDrivetrain */
//    public void set(Styles.Basic mode, double forward, double turn)
//    {
//        double l, r;
//        l = forward + turn;
//        r = forward - turn;
//
//        Drive(mode, l, r);
//    }
//    
//    public void SetVoltageRampRate(double rampRate)
//    {
//        _left.SetVoltageRampRate(rampRate);
//        _right.SetVoltageRampRate(rampRate);
//    }
//
//    public void SetVoltageCompensationRampRate(double rampRate)
//    {
//        _left.SetVoltageCompensationRampRate(rampRate);
//        _right.SetVoltageCompensationRampRate(rampRate);
//    }
//
//    public void ConfigPeakPercentOutputVoltage(double forwardVoltage, double reverseVoltage)
//    {
//        _left.ConfigPeakOutputVoltage(forwardVoltage, reverseVoltage);
//        _right.ConfigPeakOutputVoltage(forwardVoltage, reverseVoltage);
//    }
//
//    public void ConfigNominalPercentOutputVoltage(double forwardVoltage, double reverseVoltage)
//    {
//        _left.ConfigNominalOutputVoltage(forwardVoltage, reverseVoltage);
//        _right.ConfigNominalOutputVoltage(forwardVoltage, reverseVoltage);
//    }
//
//    private void Drive(Styles.Basic mode, double left, double right)
//    {
//        if (mode == Styles.Basic.Voltage)
//        {
//            _left.SetControlMode(ControlMode.BasicControlMode.kVoltage);
//            _right.SetControlMode(ControlMode.BasicControlMode.kVoltage);
//        }
//        else if (mode == Styles.Basic.PercentOutput)
//        {
//            _left.SetControlMode(ControlMode.BasicControlMode.kPercentVbus);
//            _right.SetControlMode(ControlMode.BasicControlMode.kPercentVbus);
//        }
//
//        _left.set(left);
//        _right.set(right);
//    }
//}